beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1089,7 +1074,8 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                                 Err(_) => return Err(Error::NoKzgCommitmentsFieldOnBlock),
                             };
                         if expected_kzg_commitments.is_empty() {
-                            Ok(BlobsSidecar::empty_from_parts(*block_root, block.slot()))
+                            // TODO (mark): verify this
+                            Ok(Arc::new(BlobSidecarList::empty()))
```

Comment:
this won't allocate so I think it's fine? Also better to differentiate between blobs not found and blobs not required

Reply:
@pawanjay176's pr seems to remove this entirely.. not sure why, perhaps we no longer have empty blobs?

Reply:
he has a enum that has a variant `Blob::None` (if I remember correctly)

Reply:
We no longer have to serve empty blobs via RPC (this was a relic of devnet 4) so we no longer need to do the ugly empty blob reconstruction logic. I would leave this as is and I can clean it up in pawan's PR once this is merged

Reply:
if left like this we still won't serve empty blobs, this is just an empty list. What would you do differently after pawan's?

Reply:
In that PR we're removing the fallback logic completely because it includes a db read for the block to check whether we *should* have blobs (and can return the correct `Blobs` enum variant). So I think the method just returns `None`. I don't think it was a good thing to always fallback to the block db read, at the end of the day if we don't have the blobs the data's not available. If we do want that fallback-to-block db read logic somewhere we can add it in a new method here somewhere. But I'm not sure where it's useful anymore now that we don't have to reconstruct empty blobs.